src/Dhall/Binary.hs

```diff
@@ -761,7 +762,9 @@ data DecodingFailure
     = CannotDecodeProtocolVersionString Term
     | UnsupportedProtocolVersionString Text
     | CBORIsNotDhall Term
-    deriving (Eq, Exception)
+    deriving (Eq)
+
+deriving instance Exception DecodingFailure
```

Comment:
I don't think you need `StandaloneDeriving` here.  I think this can be:

```haskell
instance Exception DecodingFailure`
```

Reply:
Oh, i followed the ghc error advice without too much thinking, i'll try that simpler option